liquidctl/driver/smart_device.py

```diff
@@ -702,6 +702,125 @@ def parse_fan_info(msg):
 
 
+class ControlHub(SmartDevice2):
+    """NZXT Control Hub fan controller.
+
+    The NZXT Control Hub is a fan controller that ships with newer NZXT cases
+    (2024+). It provides five independent fan channels with standard 4-pin
+    connectors. Both PWM and DC control is supported.
+
+    Unlike the Smart Device V2, the Control Hub uses 512-byte HID reports and
+    has 5 fan channels instead of 3. The command protocol is otherwise similar
+    to Smart Device V2.
+
+    LED control is available but uses a different protocol that is not yet
+    fully documented.
+    """
+
+    _MATCHES = [
+        (0x1e71, 0x2022, 'NZXT Control Hub', {
+            'speed_channel_count': 5,
+            'color_channel_count': 0  # LED protocol differs, not yet implemented
+        }),
+    ]
+
+    _MAX_READ_ATTEMPTS = 12
+    _READ_LENGTH = 512
+    _WRITE_LENGTH = 512
+
+    def __init__(self, device, description, speed_channel_count, color_channel_count, **kwargs):
+        """Instantiate a driver with a device handle."""
+        # Control Hub has 5 fan channels (0-4)
+        # Channel mapping (for reference, varies by case/user config):
+        #   Channel 0: typically bottom intake
+        #   Channel 1: (may be unused)
+        #   Channel 2: (may be unused)
+        #   Channel 3: typically GPU radiator or front fans
+        #   Channel 4: typically rear exhaust
+        speed_channels = {f'fan{i + 1}': (i, _MIN_DUTY, _MAX_DUTY)
+                          for i in range(speed_channel_count)}
+        # Skip SmartDevice2.__init__ and call _BaseSmartDevice directly
+        _BaseSmartDevice.__init__(self, device, description, speed_channels, {}, **kwargs)
+
+    def initialize(self, direct_access=False, **kwargs):
+        """Initialize the device and the driver.
+
+        This method should be called every time the system boots, resumes from
+        a suspended state, or if the device has just been (re)connected.
+
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        self.device.clear_enqueued_reports()
+
+        # Initialize fan reporting with V2-style command
+        self._write([0x60, 0x03])
+
+        # Request firmware info
+        self._write([0x10, 0x01])
+
+        ret = []
+
+        def parse_firm_info(msg):
+            # Firmware version at different offset for Control Hub
+            fw = f'{msg[0x11]}.{msg[0x12]}.{msg[0x13]}'
+            ret.append(('Firmware version', fw, ''))
+
+        self._read_until({b'\x11\x01': parse_firm_info})
+        return sorted(ret)
+
+    def get_status(self, direct_access=False, **kwargs):
+        """Get a status report.
+
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        ret = []
+
+        def parse_fan_info(msg):
+            # Control Hub status message format (512 bytes):
+            # Offset 15-19: mode per channel (0xff = auto, 0x00 = fixed, 0x02 = PWM)
+            # Offset 24-28: RPM values (needs verification)
```

Comment:
You can add the RPM value to the output as well.

You can call add an " (experimental)" suffix to the key name if you think it's appropriate.